src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GraphEntityDAO.scala

```diff
@@ -39,6 +39,11 @@ class GraphEntityDAO extends EntityDAO with GraphDAO {
   }
 
   private def createVertex(workspace: Vertex, workspaceNamespace: String, workspaceName: String, entity: Entity, txn: RawlsTransaction): Vertex = txn withGraph { db =>
+    // check for illegal dot characters
+    validateUserDefinedString(entity.entityType)
+    validateUserDefinedString(entity.name)
+    entity.attributes.keys.foreach(validateUserDefinedString)
+
```

Comment:
I disagree, I think the clone reason is an excellent reason not to do it.

and I think that people will want dots in their names, probably not types though.